website/src/lib/oasst_api_client.ts

```diff
@@ -0,0 +1,63 @@
+import { JWT } from "next-auth/jwt";
+
+class OasstError {
+  message: string;
+  errorCode: number;
+  httpStatusCode: number;
+
+  constructor(message: string, errorCode: number, httpStatusCode: number) {
+    this.message = message;
+    this.errorCode = errorCode;
+    this.httpStatusCode = httpStatusCode;
+  }
+}
+
+export default class OasstApiClient {
+  constructor(private readonly oasstApiUrl: string, private readonly oasstApiKey: string) {}
+
+  private async post(path: string, body: any): Promise<any> {
+    const resp = await fetch(`${this.oasstApiUrl}${path}`, {
+      method: "POST",
+      headers: {
+        "X-API-Key": this.oasstApiKey,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(body),
+    });
+
+    if (resp.status == 204) {
+      return null;
+    }
+
+    if (resp.status >= 300) {
+      try {
+        const error = await resp.clone().json();
```

Comment:
quick question, why cloning here?

Reply:
Aha! makes sense.

I guess using the `content-type` of the response is not reliable enough?

would it make sense to do it like this
```js
const text = await resp.text();
try {
  const error = JSON.parse(text);
  throw new OasstError(error.message, error.error_code, resp.status)
} catch (e) {
  // not json
  throw new OasstError(text, 0, resp.status);
}
```
this is equivalent right?

Reply:
Truth be told, I didn't think to use `content-type`. Using a `try` `catch` is a little more defensive. 

The way you've suggested is clearer and equivalent. I'll switch it!

Reply:
695e1f3